### PR TITLE
Artists module 

### PR DESCRIPTION
**Pull request by onionToyBoy**
Code style changes + renamed searchReducer + changed async storate source

### CODE REVIEW

File: MusicCatalog/index.js [hunk 1]
```diff
@@ -1,15 +1,9 @@
...
+
+import { RootNavigation } from './src/navigation/RootNavigation';
 
-import {ArtistsScreen} from './src/modules/Artists/ArtistsScreen';
-import {AlbumsScreen} from './src/modules/Albums/AlbumsScreen';
-import {ReduxProvider} from './src/store/ReduxProvider';
-import {RootNavigation} from './src/navigation/RootNavigation';
 if (__DEV__) {
-  import('./ReactotronConfig').then(() => console.log('Reactotron Configured'));
+import('./ReactotronConfig').then(() => console.log('Reactotron Configured'));
```
Comment (AnnZar12) [inline MusicCatalog/index.js]: Please, check ESLint. You need empty space before import.  Also, you could install Husky to automatically check this kind of style errors.

File: MusicCatalog/src/constants/types.js [hunk 2]
```diff
@@ -1 +1,5 @@
-export const SEARCH_ARTISTS = 'SEARCH_ARTISTS';
+export const SET_ARTISTS = 'SET_ARTISTS';
```
Comment (AnnZar12) [inline MusicCatalog/src/constants/types.js]: Filename `types` is not a suitable name for this. It's better to rename it to `actionsTypes` OR move this file to the `actions` folder.

File: MusicCatalog/src/modules/Albums/AlbumsScreen.js [hunk 3]
```diff
@@ -1,19 +1,14 @@
...
+import { SearchBar } from '../../components/SearchBar';
 
-import {colors} from '../../constants/colors';
-import {Album} from './components/Album';
+import { colors } from '../../constants/colors';
+import { Album } from './components/Album';
 
 export const AlbumsScreen = () => {
   const [isLoading, setLoading] = useState(true);
   const [data, setData] = useState([]);
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Albums/AlbumsScreen.js:10]: Note for me: This file will be updated in the scope of the next PR(Albums module).

File: MusicCatalog/src/modules/Albums/components/Album.js [hunk 4]
```diff
@@ -1,52 +1,52 @@
...
-const windowWidth = Dimensions.get('window').width;
-
 const styles = StyleSheet.create({
   container: {
     flexDirection: 'row',
     justifyContent: 'space-between',
     padding: 15,
     backgroundColor: colors.MIDDLE_GRAY,
-    width:windowWidth,
+    width: '100%',
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Albums/components/Album.js]: flex: 1

File: MusicCatalog/src/modules/Artists/actions/albumRender.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { SET_ALBUMS } from '../../../constants/types';
+
+export const albumRender = albums => {
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/actions/albumRender.js]: It's better to call this function `setAlbums`. Also, you don't need to create a separate file for each action. You can just create an index.js file in the actions folder and paste all module's actions here.

File: MusicCatalog/src/modules/Artists/actions/artistRender.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { SET_ARTISTS } from '../../../constants/types';
+
+export const artistRender = artists => {
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/actions/artistRender.js]: setArtists

File: MusicCatalog/src/modules/Artists/actions/trackRender.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { SET_TRACKS } from '../../../constants/types';
+
+export const trackRender = tracks => {
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/actions/trackRender.js]: setTracks

File: MusicCatalog/src/modules/Artists/components/AlbumTracks.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+import { getTracks } from '../thunks';
+import { Track } from './Track';
+
+export const AlbumTracks = ({ componentId }) => {
+  const dispatch = useDispatch();
+
+  const tracks = useSelector(selectTracks);
+
+  useEffect(() => {
+    dispatch(getTracks(componentId));
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/components/AlbumTracks.js]: `getTracks` by navigation screen ID value is not a good idea. It's better to pass prop `albumId` to the component.

File: MusicCatalog/src/modules/Artists/components/AlbumTracks.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  }, [dispatch]);
+
+  const renderTracks = ({ item }) => (
+    <Track name={item.trackName} number={item.trackNumber} time={item.trackTimeMillis} />
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/components/AlbumTracks.js]: FYI: You can pass this params as `<Track  {...item} />` or `<Track track={item} />`. The current option is acceptable, but sometimes it's hard to read if you pass a lot of values.

File: MusicCatalog/src/modules/Artists/components/ArtistsAlbums.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  };
+
+  const renderAlbums = ({ item }) => (
+    <Album
+      id={item.collectionId}
+      onOpenTracks={onOpenTracks}
+      artistName={item.artistName}
+      collectionName={item.collectionName}
+      cover={item.artworkUrl60}
+      collectionPrice={checkPrice(item.collectionPrice)}
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/components/ArtistsAlbums.js]: It's better to pass the `collectionPrice` value and call `checkPrice` function before the render.

File: MusicCatalog/src/modules/Artists/components/ArtistsAlbums.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+import { checkPrice } from '../../../utils/checkPrice';
+import { routes } from '../../../constants/routes';
+
+export const ArtistsAlbums = ({ componentId }) => {
+  const dispatch = useDispatch();
+
+  const albums = useSelector(selectAlbums);
+
+  useEffect(() => {
+    dispatch(getAlbums(componentId));
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/components/ArtistsAlbums.js]: The same comments as for the AlbumTracks screen

File: MusicCatalog/src/modules/Artists/components/ArtistsScreen.js [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+  const onOpenAlbum = (name, id) => {
+    Navigation.push(componentId, {
+      component: {
+        name: routes.ArtistsAlbums,
+        id: id,
+        options: {
+          topBar: {
+            title: {
+              text: name + ' albums',
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/components/ArtistsScreen.js]: You can also use template strings  \`${name} albums\`

File: MusicCatalog/src/modules/Artists/components/Track.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+export const Track = ({ name, number, time }) => {
+  return (
+    <TouchableOpacity style={styles.container}>
+      <View style={styles.leftContainer}>
+        <Text style={styles.numbers}>{number}</Text>
+        <Text style={styles.name}>{name}</Text>
+      </View>
+      <View style={styles.rightContainer}>
+        <Text style={styles.numbers}>{timeConventer(time)}</Text>
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/components/Track.js]: Please, move timeConventer(time) to constants

File: MusicCatalog/src/modules/Artists/components/Artist.js [hunk 14]
```diff
@@ -1,22 +1,19 @@
...
-} from 'react-native';
+import { View, StyleSheet, TouchableOpacity, Text } from 'react-native';
 
-import {colors} from '../../../constants/colors';
-import {symbols} from '../../../constants/symbols';
+import { colors } from '../../../constants/colors';
+import { symbols } from '../../../constants/symbols';
+
+export const Artist = ({ name, genre, onOpenAlbum, id }) => {
+const onPressArtist =() => onOpenAlbum(name, id)
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/components/Artist.js]: Please, check ESLint and code style

File: MusicCatalog/src/modules/Artists/reducers/tracks.js [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import { SET_TRACKS } from '../../../constants/types';
+
+const INITIAL_STATE = {
+  tracks: {},
```
Comment (AnnZar12) [inline MusicCatalog/src/modules/Artists/reducers/tracks.js]: Please, organize the logic to save trucks for different artists.

File: MusicCatalog/src/utils/checkPrice.js [hunk 16]
```diff
@@ -0,0 +1 @@
+export const checkPrice = price => (price>0 ? price + ' $' : 'FREE');
```
Comment (AnnZar12) [inline MusicCatalog/src/utils/checkPrice.js]: You don't need to create a separate file for each helper. Please, create an index.js file and move all utils functions here.

**Review by AnnZar12 — CHANGES_REQUESTED** [review]

**Comment by onionToyBoy** [comment]
Artists module

**Pull request merged**